Add marine equipment specialist document

minhngocd opened the pull request:
New specialist publisher for Marine Equipment Approved Recommendations by Maritime Coastguard Agency.

minhngocd committed:
[1141] - Added new specialist publisher for Marine Equipment Approved Recommendations

Maritime Coastguard Agency requested a new specialist publisher and finder for UK Marine Equipment Approved Recommendations.
Added required models, views and schemas for publishing and managing the new document type.
The fields Year Adopted, Keyword, and Reference Number are set to non-filterable. Original requirements asked for them to be filterable.
However, due to Elasticsearch having to be able to reindex over new free text values, and they are non aggregated fields, meaning filtering on free text is currently not supported functionality on search-api.
There is an existing search box to free text search to serve lookup requirements.

minhngocd committed:
[1141] - Removed validation for presence of facets for marine equipment approved recommendations publisher

The requirements document supplied by Maritime Coastguard Agency specified them to be optional fields.

baisa commented on lib/documents/schemas/marine_equipment_approved_recommendations.json:
```diff
@@ -0,0 +1,102 @@
+      "filterable": true,
+      "allowed_values": [
+        {
+          "label": "Life-saving appliances",
```
Should we have a dash in `Life-saving`?

minhngocd replied:
This came from their requirements documentation - we can double check with MCA when we run a test in integration?

baisa commented on lib/documents/schemas/marine_equipment_approved_recommendations.json:
```diff
@@ -0,0 +1,102 @@
+          "value": "navigation-equipment"
+        },
+        {
+          "label": "Radio-communication equipment",
```
Similarly here

minhngocd replied:
Same, it came from their requirements documentation - we can double check with MCA when we run a test in integration?

baisa reviewed (commented):
This is looking good! Just added minor comments. We could consider commits cleanup - as removing validation isn’t separate logic bit, we could probably include in in the first commit. Pre production flag commits could be also merged into one.

minhngocd committed:
[1141] - Add show summaries flag for marine equipment specialist finder

Requirements requested finder to show summaries for each entry.

minhngocd committed:
[1141] Added email filters for marine equipment signup

As part of MCA's requirements, emails are supposed to be filtered via categories
